src/structures/APIMessage.js

```diff
@@ -151,8 +151,8 @@ class APIMessage {
 
     let nonce;
     if (typeof this.options.nonce !== 'undefined') {
-      nonce = parseInt(this.options.nonce);
-      if (isNaN(nonce) || nonce < 0) throw new RangeError('MESSAGE_NONCE_TYPE');
+      nonce = typeof this.options.nonce === 'string' ? this.options.nonce : parseInt(this.options.nonce);
+      if (isNaN(nonce) && !nonce) throw new RangeError('MESSAGE_NONCE_TYPE');
```

Comment:
If we're going to go strictly for what Discord accepts, the `nonce` field accepts either a string or an integer.

The valid way to check for integer is to use [`Number.isInteger()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger).

The data validation here is very loose, and should be applied only for integers, avoiding further processing on strings.

Please correct me if I'm wrong.

cc: @vladfrangu

Reply:
The original code was converting param with `parseInt` I also agree check if Number is Integer

Reply:
I would like to commit it

Reply:
I agree with @kyranet here, the check for number should be done only if the nonce IS a number

Reply:
Please check the new commit!